### DISCUSSION TITLE
Pan/Scrolling while dragging not react-flow node

### DISCUSSION THREAD

**Discussion by antollika** [xyflow/xyflow#4580] [Q&A] [answered]
Hello

We have `p̶a̶n̶O̶n̶S̶c̶r̶o̶l̶l̶` -> `panOnDrag` prop and it works fine when I'm dragging react-flow nodes around the canvas.

Is it possible to achieve the same behavior while dragging NOT react-flow node?

React flow examples page has this case: [Drag and Drop](https://reactflow.dev/examples/interaction/drag-and-drop)
We can use HTML or 3rd party library to drag and drop items to react-flow canvas.
But - when I'm dragging the item near the area sides -> pan does not work.

Hope for your help

**Comment by bcakmakoglu** [comment, marked as answer]
`panOnScroll` has nothing to do with panning the viewport when dragging a node.
That's [`autoPanOnNodeDrag`](https://reactflow.dev/api-reference/react-flow#auto-pan-on-node-drag)

It's possible to achieve the same behavior but there aren't any *direct* helpers that would do this.
Either you implement something similar or, whenever you drag a node over the viewport create a "ghost" node (like an invisible or opaque one) that acts as a sort of placeholder until the drop is finished, which would allow the viewport to be panned via the ghost node.

**Reply by antollika** [reply]
You are right -> I mean `panOnDrag`

"ghost" node creation - sounds interesting but it's quite a chore.

Thank you for your advice
